lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -250,31 +250,47 @@ class MongoClientInterface {
     }
 
     getBucketAndObject(bucketName, objName, params, log, cb) {
-        this.getBucketAttributes(bucketName, log, (err, bucket) => {
-            if (err) {
-                log.error(
-                    'getBucketAttributes: error getting bucket attributes',
-                    { error: err.message });
-                return cb(err);
-            }
-            this.getObject(bucketName, objName, params, log, (err, obj) => {
-                if (err) {
-                    if (err === errors.NoSuchKey) {
-                        return cb(null,
-                                  { bucket:
-                                    BucketInfo.fromObj(bucket).serialize(),
-                                  });
+        return async.parallel([
```

Comment:
Just a detail, but naming the operations (using a map instead of array) allows for better readability:
```js
async.parallel({
    obj: callback => {...},
    bucket: callback => {...},
}, (err, results) => {
    const { obj, bucket } = results;
    ...
});
```